CalcualtorLogic.js

```diff
@@ -26,7 +26,7 @@ calculatorLogic.divide();
 class Calculator {
     currentResult = 0;
     resultElement = null;
-    operator = this.currentResult;
+    operator = this.buttonElement;
```

Comment:
what is `this.buttonElement` ?

Reply:
refers to the clicked operator (+||-||*||/)

 const div = document.createElement("div");
        for (let button of buttons) {
            var buttonElement = document.createElement("button");
            buttonElement.dataset.number = button;
            buttonElement.classList.add("button");
            const buttonElementText = document.createTextNode(button);
            buttonElement.appendChild(buttonElementText);
            div.appendChild(buttonElement);
            buttonElement.addEventListener("click", this.onButtonNumberClick);
        }
        return div;
    }

Reply:
First `buttonElement` is not defined anywhere in the class 
second, when I open the calculator for the first, I didn't click on anything yet. so does it make sense that you are trying to assign a value for the property 

in the previous implantation of the calculator I can see that you set `null` as value for the `operator` variable 
` let operator = null;`
it make sense to set its value to null here also.

then in click event handler of the operators, you update this property with the value of the clicked button

Reply:
but in the getButtonRow method we have ::
var buttonElement = document.createElement("button");